builtin/logical/pki/path_revoke.go

```diff
@@ -461,26 +479,23 @@ func validatePrivateKeyMatchesCert(signer crypto.Signer, certReference *x509.Cer
 	// See: https://github.com/golang/go/blob/c6a2dada0df8c2d75cf3ae599d7caed77d416fa2/src/crypto/tls/tls.go#L304-L331
 	switch certPub := certReference.PublicKey.(type) {
 	case *rsa.PublicKey:
-		privPub, ok := signer.Public().(*rsa.PublicKey)
+		privPub, ok := verifier.(*rsa.PublicKey)
 		if !ok {
 			return errutil.UserError{Err: "provided private key type does not match certificate's public key type"}
```

Comment:
this is now under validatePublicKeyMatchesCert - the error message refers to a private key.

Reply:
`validatePrivateKeyMatchesCert` calls `validatePublicKeyMatchesCert` and most callers outside of ACME are using an explicit private key. Here, in ACME, we're validating the private key by proxy (JWS signature over the message).

Reply:
I've updated the `acmeRevocationByPoP` error message to say `unable to verify PoP of private key provided by proxy: ` to clarify to the caller why private key is mentioned elsewhere in the error message.